components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,232 @@
+import styled from "styled-components";
+import Link from "next/link";
+
+export default function CreateOrUpdateRecipeForm({
+  errorMessage,
+  inputValidation,
+  recipeDetails = {},
+  onSubmit,
+}) {
+  const formTitle =
```

Comment:
A potentially cleaner solution would be to check if recipeDetails has no keys once. 

i.e. 

``` js
const isCreatingNewRecipe = Object.keys(recipeDetails).length === 0;

const formTitle = isCreatingNewRecipe ? "Create new recipe" : "Update Recipe";
const formButtonTitle = isCreatingNewRecipe ? "Add Recipe" : "Save Changes";
const abortLinkUrl = isCreatingNewRecipe ? "/" : `/recipe/${recipeDetails.id}`;
```